core/src/main/java/org/apache/calcite/util/BuiltInMethod.java

```diff
@@ -586,7 +587,11 @@
       "resultSelector", Function2.class),
   AGG_LAMBDA_FACTORY_ACC_SINGLE_GROUP_RESULT_SELECTOR(AggregateLambdaFactory.class,
       "singleGroupResultSelector", Function1.class),
-  TUMBLING(EnumerableDefaults.class, "tumbling", Enumerable.class, Function1.class);
+  TUMBLING(EnumerableDefaults.class, "tumbling", Enumerable.class, Function1.class),
+  HOPPING(EnumUtils.class, "hopping", Enumerator.class, int.class, long.class,
```

Comment:
Any special reasons to implement `tumbling` and `hopping`&`sessionize` in different places?

Reply:
I see. No special reason. Would you suggest I put it in the same class? If so, which class is better?

Reply:
Current now, I personally suggest to put them in `EnumUtils`. It is a little strange in Linq4j. But if we will add more streaming functions later, I think we can create a separate class for them. It's more clear.

Reply:
Done!